### PR TITLE
Discriminator Class without sklearn Dependency

### PR DESCRIPTION
**Pull request by sumitpuri**
### Summary
The current Discriminator code uses LDA and QDA from sklearn library. There is a need to remove this dependency of the code on sklearn library. This PR adds the logic that performs the LDA and QDA without sklearn library.

### Details and comments
A new python module called discriminator_nosklearn is introduced that includes two classes: LDAnosklearn and QDAnosklearn. Using this module, there will be no dependency on the sklearn package while performing the discriminator.

### CODE REVIEW

File: qiskit_experiments/measurement/discriminator/discriminator_nosklearn.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import numpy as np
+from numpy.linalg import inv
+from math import log
+
+    
+
+class LDAnosklearn():
+    
+    def __init__(self):
```
Comment (coruscating) [inline qiskit_experiments/measurement/discriminator/discriminator_nosklearn.py:14]: please add docstrings to all methods

File: qiskit_experiments/measurement/discriminator/discriminator_nosklearn.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        return None
+    
+    def fit(self, X, y):
+        X0 = X[y==0]
```
Comment (coruscating) [inline qiskit_experiments/measurement/discriminator/discriminator_nosklearn.py:35]: this needs comments and citation for where the formula came from

File: qiskit_experiments/measurement/discriminator/discriminator_nosklearn.py (hunk 1 above)
Comment (sumitpuri) [reply qiskit_experiments/measurement/discriminator/discriminator_nosklearn.py:14]: Added docstring

File: qiskit_experiments/measurement/discriminator/discriminator_nosklearn.py (hunk 2 above)
Comment (sumitpuri) [reply qiskit_experiments/measurement/discriminator/discriminator_nosklearn.py:35]: Added citation

**Pull request closed without merging**